src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -167,9 +156,8 @@ - (void)cancel {
 }
 
 - (void)dealloc {
-  grpc_call *gRPCCall = _gRPCCall;
   dispatch_async(_callQueue, ^{
-    grpc_call_destroy(gRPCCall);
+    _wrappedCall = nil;
```

Comment:
To be able and set the ivar, the block retains a reference to `self` (this line is equivalent to `self->_wrappedCall = nil;`). We don't want this, because we're in the middle of deallocating self :) So when the block runs anything could be in the position in memory where we're writing `nil` (and sending `-release`!). What we want here is to retain only _wrappedCall. To do that, use the same pattern we had with `_gRPCCall`:
```objective-c
GRPCWrappedCall *wrappedCall = _wrappedCall;
dispatch_async(_callQueue, ^{
  wrappedCall = nil;
});
```